### DISCUSSION TITLE
Image registration using label mask

### DISCUSSION THREAD

**Discussion by Tibetan-eagle** [SuperElastix/elastix#1258] [General]
Hi Developers, 

I have an issue that has been raised similarly in other discussions but I think the workflow is slightly different so wanted to get this discussion going. 

Basically, I converted the dicom (CT, CBCT, RTStructure) into .mha files using plastimatch. Then I feed those to do deformable registration using the parameters in the Elastix manual for now. It seems to be working ok, but now I want to focus on specific structures. I have contours of the planning CT, which has already be converted to Bladder.mha, Prostate.mha for instance and after running the elastix command (./elastix \ -f /data/work/ProstateSeg/Elastix-test/Converted_data/CBCT1.mha -m /data/work/ProstateSeg/Elastix-test/Converted_data/CT.mha -mMask /data/work/ProstateSeg/Elastix-test/Converted_data/Bladder.mha -out /data/work/ProstateSeg/Elastix-test/Registered_data -p /data/work/ProstateSeg/Elastix-test/parameter.txt)

I get the following error: itk::ExceptionObject (0x55ff01bc8690)
Location: "ElastixTemplate - Run()" 
File: /home/niels/dev/src/elastix/Common/CostFunctions/itkAdvancedImageToImageMetric.hxx
Line: 1007
Description: itk::ERROR: AdvancedMattesMutualInformationMetric(0x55ff01bacf40): Too many samples map outside moving image buffer: 27 / 45056

Any solution this? I am using Elastix 5.0.0 (downloaded release version).

**Comment by mstaring** [comment]
Hi tenzin,

Did you see this page?
https://github.com/SuperElastix/elastix/wiki/FAQ

Perhaps the initialization is not sufficient for the registration to catch on.


________________________________
From: Tenzin K ***@***.***>
Sent: 28 October 2024 17:46
To: SuperElastix/elastix ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [SuperElastix/elastix] Image registration using label mask (Discussion #1258)


Hi Developers,

I have an issue that has been raised similarly in other discussions but I think the workflow is slightly different so wanted to get this discussion going.

Basically, I converted the dicom (CT, CBCT, RTStructure) into .mha files using plastimatch. Then I feed those to do deformable registration using the parameters in the Elastix manual for now. It seems to be working ok, but now I want to focus on specific structures. I have contours of the planning CT, which has already be converted to Bladder.mha, Prostate.mha for instance and after running the elastix command (./elastix \ -f /data/work/ProstateSeg/Elastix-test/Converted_data/CBCT1.mha -m /data/work/ProstateSeg/Elastix-test/Converted_data/388203_CT.mha -fMask /data/work/ProstateSeg/Elastix-test/Converted_data/388203_Bladder.mha -out /data/work/ProstateSeg/Elastix-test/Registered_data -p /data/work/ProstateSeg/Elastix-test/parameter.txt)

I get the following error: itk::ExceptionObject (0x55ff01bc8690)
Location: "ElastixTemplate - Run()"
File: /home/niels/dev/src/elastix/Common/CostFunctions/itkAdvancedImageToImageMetric.hxx
Line: 1007
Description: itk::ERROR: AdvancedMattesMutualInformationMetric(0x55ff01bacf40): Too many samples map outside moving image buffer: 27 / 45056

Any solution this? I am using Elastix 5.0.0 (downloaded release version).
